### ISSUE TITLE
Support single node clusters

### ISSUE THREAD

**Issue by PascalTurbo** [Opster/opensearch-k8s-operator#233]
On productive systems it's common to have large clusters. But when only testing a setup, installing OpenSearch for experimentation or education installing a full cluster will wast a lot of resources what can be really expensive when working on f.e. cloud services. For this case it would be great to have the smallest possible setup available.

**Comment by mrvdcsg** [comment]
I agree with what is posted above. I will add that there are additional benefits when you do localized/edge cross-cluster-replicated nodes with a service mesh that fails over to a cloud instance. You may want a single node cluster at the edge to fit on extremely limited hardware. This is possible with helm charts but not when using the operator. I'd like to see this. capability added.

**Comment by RiRa12621** [comment]
Is this up for debate or up for implementation?

**Comment by swoehrl-mw** [comment]
Hi @RiRa12621. To be honest, I don't really see this as a worthwhile feature to support. The operator is geared towards managing multi-node HA opensearch clusters and I fear having logic for single-node clusters could complicate or interfer core logic.
That said, if someone contributes a good implementation, I will not actively block it.

**Comment by pasztorl** [comment]
there are tasks for single instance too so operator can be a great help on it, like doing backups/restores and certificate management. we use 3 node clusters and indices replicated on each of them. it would be awesome if we can create a single instance for the testing envs using backups from these cluster using operator. Of course we can do this by hand, but this is an example where operator can simplify the workflows.

**Comment by estromenko** [comment]
Hello everyone! I tried to implement a simple solution using extra environment variables in operator (`cluster.initial_master_nodes` and `discovery.type`), seems like it's working. @swoehrl-mw should I create a PR for this? Or can we discuss another solution for this feature?

**Comment by lucasfcnunes** [comment]
@estromenko
Let's go!

*Standalone, 1 node/instance "cluster" for development purposes is what I am trying to come up with right now.

Setting to just 1 replica makes it get stuck on the following log:

```jsonl
{"type": "json_logger", "timestamp": "2025-04-04T21:23:39,950Z", "level": "INFO", "component": "o.o.s.c.ConfigurationRepository", "cluster.name": "opensearch-cluster", "node.name": "opensearch-cluster-masters-0", "message": "Wait for cluster to be available ..." } 
```

**Comment by galovics** [comment]
Hi team, is this going to be solved in the near future?

**Comment by synhershko** [comment]
Not in the near future @galovics

**Comment by galovics** [comment]
@synhershko noted, thanks for the update.
